### PR TITLE
authdb: allow admin cap to update+delete user api key

### PR DESCRIPTION
**Pull request by sluongng**
Modify authdb.go so that credentials with admin capability
can Update and Delete user's api keys.

### CODE REVIEW

File: enterprise/server/backends/authdb/authdb.go [hunk 1]
```diff
@@ -967,16 +967,19 @@ func (d *AuthDB) authorizeAPIKeyWrite(ctx context.Context, h interfaces.DB, apiK
 	if err != nil {
 		return nil, err
 	}
+	err = authutil.AuthorizeOrgAdmin(user, key.GroupID)
+	if err == nil {
+		// admins have write permissions on all keys in the group.
+		return key, nil
+	}
```
Comment (bduffany) [inline enterprise/server/backends/authdb/authdb.go:974]: `AuthorizeOrgAdmin` is not quite what we want here - we only want to allow API keys with ORG_ADMIN capability. See the authdb changes in https://github.com/buildbuddy-io/buildbuddy/pull/7666

File: enterprise/server/backends/userdb/userdb_test.go [hunk 2]
```diff
@@ -966,17 +964,23 @@ func TestUserOwnedKeys_GetUpdateDeletePermissions(t *testing.T) {
...
 			if test.Accessor != "" {
 				accessorCtx = authUserCtx(accessorCtx, env, t, test.Accessor)
 			}
-			keys, err := adb.GetUserAPIKeys(accessorCtx, test.Accessor, ownerGroup.GroupID)
-			// Only the owner should be able to view or update the API key,
-			// regardless of role.
-			isAuthorized := test.Owner == test.Accessor
+			keys, err := adb.GetUserAPIKeys(accessorCtx, test.Owner, ownerGroup.GroupID)
+			// Only the owner and group's admin should be able to view or update the API key.
+			isAuthorized := test.Owner == test.Accessor || test.Accessor == groupAdminID
```
Comment (bduffany) [inline enterprise/server/backends/userdb/userdb_test.go:983]: I mentioned in the other comment but this isn't the behavior we want - we want to check ORG_ADMIN capability on the authenticated API key, not group admin role on the authenticated user.

The test code exercising ORG_ADMIN accessing personal keys is here: https://github.com/buildbuddy-io/buildbuddy/blob/3fd4dd9e23fb21995f6db370803d208a269ee8f7/enterprise/server/backends/userdb/userdb_test.go#L1410-L1431

It's already testing Create/List/Read functionality; it just needs Update/Delete tests to be added.

**Review by bduffany — CHANGES_REQUESTED** [review]

File: enterprise/server/backends/authdb/authdb.go (hunk 1 above)
Comment (sluongng) [reply enterprise/server/backends/authdb/authdb.go:974]: I was conflating admin role and admin capability. 🤦 

Will fix.

File: enterprise/server/backends/userdb/userdb_test.go (hunk 2 above)
Comment (sluongng) [reply enterprise/server/backends/userdb/userdb_test.go:983]: I changed the implementation but this test would still require an update to check for the admin user. I did add the other test to validate admin capability from an API key though.

**Review by bduffany — APPROVED** [review]

**Comment by sluongng** [comment]
Since this is auth-related code, gona merge after another +1 review

**Pull request merged**